crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -951,10 +949,46 @@ impl EntityCommands<'_> {
     /// # bevy_ecs::system::assert_is_system(add_combat_stats_system);
     /// ```
     #[track_caller]
-    pub fn insert(&mut self, bundle: impl Bundle) -> &mut Self {
+    pub fn insert(self, bundle: impl Bundle) -> Self {
         self.add(insert(bundle, InsertMode::Replace))
     }
 
+    /// Similar to [`Self::insert`] but will only insert if the predicate returns true.
```

Comment:
```suggestion
    /// Similar to [`Self::insert`] but will only insert if the predicate returns true.
    /// This is useful for chaining method calls.
```
This comment is also valid for the other methods